### DISCUSSION TITLE
Multiple instances per plugin

### DISCUSSION THREAD

**Discussion by SebastianStehle** [danny-avila/LibreChat#1843] [Feature Requests & Suggestions]
Hi,

we have a fork of librechat where we basically do two things: 

1. Show a disclaimer text somwhere.
2. Configure custom plugins.

The plugins itself are just copy and paste from existing plugins to fetch data from Azure AI Search or to call an API in the future. This is not a good reason to maintain a fork in my opinion. So ideally there would be one of the following feature:

1. Having the ability to define multiple plugins via environment variables. We could also use a configuration library that supports multiple configuration sources (like viper in golang of Microsoft.Extensions.Configuration in .NET).

2. Having and admin system where you can add plugins yourself and store the config in MongoDB. Not sure how to deal with the secrets there, perhaps just an optional encryption key, that can be injected via the config system. Futhermore a general "tool" extension that just invokes a http endpoint would be great.

At the moment I have plenty of time at work and can help with that. This can change anytime ofc, so I don't want to make promises

**Comment by ndrsfel** [comment]
+1 on global plugins, I like the idea. In general, default settings like enabled plugins and system prompts for new users would be awesome
